### PR TITLE
DAOS-14220 test: Fix ior errors management and fetch metric threshold #12478 #13122

### PR DESCRIPTION
**Pull request by knard-intel**
### Description

This PR is clean cherry-pick of the two PRs #12478 and #13122.

When the ior command is failing, it is not possible to properly test the metrics as we do not know how much data have been transferred.  When such error is occurring, the best option is to fail the test.
The `engine_pool_xferred_fetch` metric threshold is updated to manage extra networking control traffic.
Some comments are added on the limitation of the tests.

### Validation

Test-Tag pragma is used to only run functional tests which could be impacted by this PR.

### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by shimizukko — APPROVED** [review]

**Review by daltonbohning — APPROVED** [review]

**Review by daltonbohning — CHANGES_REQUESTED** [review]
Ticket does not have merge approval:
https://daosio.atlassian.net/browse/DAOS-14220
But otherwise looks good to land. Please re-add gatekeeper after getting approval

**Comment by knard-intel** [comment]
> Ticket does not have merge approval: https://daosio.atlassian.net/browse/DAOS-14220 But otherwise looks good to land. Please re-add gatekeeper after getting approval

The approved to merge is OK for the two tickets.
Thanks to have show me that I forgot one of them.

**Review by daltonbohning — APPROVED** [review]

**Pull request merged**
